### PR TITLE
docs: Update arch doc for one proxy / VM

### LINKED ISSUE #908: Is there only one cc-proxy instance running per Clear Containers host?
I get the follow description about cc-pproxy in [architecture.md](https://github.com/clearcontainers/runtime/blob/master/docs/architecture/architecture.md)
```
The Clear Containers proxy (cc-proxy) multiplexes and demultiplexes those commands and streams for all container virtual machines. There is only one cc-proxy instance running per Clear Containers host.
```

But I found that each container has a corresponding `cc-proxy` process when I create two containers by docker+cc on my host.  Have anything error in my env? 

image

Expect to get help, thanks!

### PR DESCRIPTION
**Pull request by jodh-intel**
Update the architecture document to reflect the new proxy behaviour
where there is a proxy instance per virtual machine rather than a
single host-level proxy instance.

This should strictly have been handled on #835.

Fixes #908.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
Hi @sameo, @sboeuf, @grahamwhaley - could you take a look?

Note also: https://github.com/clearcontainers/proxy/issues/193.

**Comment by sboeuf** [comment]
LGTM

File: docs/architecture/architecture.md [hunk 1]
```diff
@@ -393,24 +396,20 @@ for single container pods.
...
-to multiple `cc-shim` and `cc-runtime` clients. Only a single instance of `cc-proxy`
-per host is necessary as it can be used for several different VMs.
-Its main role is to:
-- Arbitrate access to the `cc-agent` control channel between all the `cc-runtime`
-  instances and the `cc-shim` ones.
-- Route the I/O streams and signals between the various `cc-shim` instances and
-  the `cc-agent`.
-
-The `cc-proxy` API is available through a single socket for all `cc-shim` and
+to a single `cc-shim` and `cc-runtime` client. Its main role is to route the
```
Comment (sameo) [inline docs/architecture/architecture.md]: A **single** `cc-shim`?

Comment (jodh-intel) [reply docs/architecture/architecture.md]: Good catch! Branch updated.

**Comment by jodh-intel** [comment]
Hi @klynnrif - please could you my changes [*]?

---

[*] - If there are other parts of the doc that need altering, let's do that on a separate PR as this doc is a bit of a monster :smile:

File: docs/architecture/architecture.md [hunk 2]
```diff
@@ -273,15 +273,18 @@ When handling the OCI `create` command, `cc-runtime` goes through the following
 4. Create and start the virtual machine that will run the container process. The
   VM will run inside the host namespaces created during step 1, and its `systemd`
   instance will spawn the `cc-agent` daemon.
-5. Register the virtual machine with `cc-proxy`.
-6. The `cc-proxy` waits for the agent to signal that it is ready and then returns
+5. Spawn the `cc-proxy` process providing a single argument:
+  `cc-proxy --uri $(uri)`
+   * A UNIX socket URI, which will be used by the `cc-shim` and `cc-agent` processes will use to pass information between them.
```
Comment (grahamwhaley) [inline docs/architecture/architecture.md]: I think s/will use// ?

File: docs/architecture/architecture.md [hunk 3]
```diff
@@ -273,15 +273,18 @@ When handling the OCI `create` command, `cc-runtime` goes through the following
...
+6. Register the virtual machine with `cc-proxy`.
+7. The `cc-proxy` waits for the agent to signal that it is ready and then returns
   a token. This token uniquely identifies a process within a container inside
   the virtual machine.
-7. Spawn the `cc-shim` process providing two arguments:
+8. Spawn the `cc-shim` process providing two arguments:
```
Comment (grahamwhaley) [inline docs/architecture/architecture.md]: Was surprised you did not change these all to (1.) items whilst here @jodh-intel:-)

File: docs/architecture/architecture.md [hunk 4]
```diff
@@ -333,8 +336,8 @@ goes through the following steps:
   to either `containerd-shim` in the Docker use cases or `conmon` for the Kubernetes
   deployements.
 4. `cc-runtime` sends an agent `EXECMD` command to start the command in the
-  right container. The command is sent to `cc-proxy` who forwards it to the right
-  agent instance running in the appropriate guest.
+  container. The command is sent to `cc-proxy` who forwards it to the
```
Comment (grahamwhaley) [inline docs/architecture/architecture.md]: s/who/which/

File: docs/architecture/architecture.md (hunk 3 above)
Comment (jodh-intel) [reply docs/architecture/architecture.md]: I considered it, but then decided to go for a minimal change "in line" with the current content to minimise the review cycle time ;)

File: docs/architecture/architecture.md (hunk 4 above)
Comment (jodh-intel) [reply docs/architecture/architecture.md]: However, branch updated :smile:

Comment (klynnrif) [inline docs/architecture/architecture.md]: Suggested rewrite: the command is sent to `cc-proxy`, which forwards it to the

File: docs/architecture/architecture.md [hunk 5]
```diff
@@ -393,24 +396,21 @@ for single container pods.
   instances, `cc-runtime` waits for all of them to terminate as well.
 4. `cc-runtime` sends the `UnregisterVM` command to `cc-proxy`, to let it know
   that the virtual machine that used to host the pod should no longer be used.
-5. `cc-runtime` explicitly shuts the virtual machine down.
-6. The host namespaces are cleaned up and destroyed. In particular, `cc-runtime`
+5. The `cc-proxy` instance will then exit.
```
Comment (klynnrif) [inline docs/architecture/architecture.md]: The `cc-proxy` instance exits.

File: docs/architecture/architecture.md [hunk 6]
```diff
@@ -393,24 +396,21 @@ for single container pods.
...
+6. `cc-runtime` explicitly shuts the virtual machine down.
```
Comment (klynnrif) [inline docs/architecture/architecture.md]: `cc-runtime` shuts the virtual machine down.

File: docs/architecture/architecture.md [hunk 7]
```diff
@@ -393,24 +396,21 @@ for single container pods.
...
+7. The host namespaces are cleaned up and destroyed. In particular, `cc-runtime`
   offloads the networking namespace cleanup path by calling into the specific
   networking model (CNM or CNI) removal method.
-7. All remaining pod related resources on the host are deleted.
+8. All remaining pod related resources on the host are deleted.
```
Comment (klynnrif) [inline docs/architecture/architecture.md]: All remaining pod-related resources...

**Comment by jodh-intel** [comment]
Hi @klynnrif - thanks for reviewing - branch updated.

**Pull request merged**